spacy/cli/project/run.py

```diff
@@ -308,3 +317,37 @@ def get_fileinfo(project_dir: Path, paths: List[str]) -> List[Dict[str, Optional
         md5 = get_checksum(file_path) if file_path.exists() else None
         data.append({"path": path, "md5": md5})
     return data
+
+
+def _validate_requirements(requirements: List[str]) -> Tuple[bool, bool]:
+    """Checks whether requirements are installed and free of version conflicts.
+    requirements (List[str]): List of requirements.
+    RETURNS (Tuple[bool, bool]): Whether (1) any packages couldn't be imported, (2) any packages with version conflicts
+        exist.
+    """
+
+    failed_pkgs: List[str] = []
+    conflicting_pkgs_msgs: List[str] = []
+
+    for req in requirements:
+        try:
+            pkg_resources.require(req)
+        except pkg_resources.DistributionNotFound:
+            failed_pkgs.append(req)
+        except pkg_resources.VersionConflict as vc:
+            conflicting_pkgs_msgs.append(vc.report())
+
+    if len(failed_pkgs):
+        msg.warn(
+            f"The following requirements could not be validated:\n{failed_pkgs}.",
+            "Make sure your Python environment is set up correctly and you installed all requirements specified in "
+            "your project's requirements.txt.",
+        )
+    if len(conflicting_pkgs_msgs):
+        msg.warn(
+            f"The following dependency conflicts were detected:\n{conflicting_pkgs_msgs}",
+            "Make sure your Python environment is set up correctly. Ensure that your project's "
+            "requirements.txt specifies the correct versions for all dependencies.",
```

Comment:
This is also confusing when the "conflicting" things are the actual requirements vs. conflicts further down in the dependency tree. For example:

```
⚠ The following requirements could not be validated:
['spacy-streamlit==1.0.2'].
Make sure your Python environment is set up correctly and you installed all
requirements specified in your project's requirements.txt.
⚠ The following dependency conflicts were detected: ['spacy-loggers
1.0.3 is installed but spacy-loggers==1.0.2 is required']
Make sure your Python environment is set up correctly. Ensure that your
project's requirements.txt specifies the correct versions for all dependencies.
```

This message is kind of repeated twice and the incorrect version isn't exactly the same thing as other kinds of conflicts? Maybe there's no way to distinguish this, but I think at least the message is confusing.

Reply:
Should we ditch the message completely? Re-reading it now makes me think it should be obvious to check your env setup/`requirements.txt` if warnings about requirements/dependencies show up.

Regarding conflicts with actual requirements vs. further down the tree: it's easy enough to rephrase the `pkg_resources` output, but I'd keep the original phrasing for consistency.

Reply:
I think the first message is clearer ("Make sure your Python environment is set up correctly and you installed all requirements specified in your project's requirements.txt.") and would like to just see it once even if there are both types of errors.

Reply:
Combined both into a slightly modified error message that's only shown once. Example output:
```
⚠ Missing requirements or requirement conflicts detected. Make sure
your Python environment is set up correctly and you installed all requirements
specified in your project's requirements.txt. Affected:
["The 'scipy' distribution was not found and is required by the application",
"The 'transformers<4.3.0,>=3.4.0' distribution was not found and is required by
the application", "The 'spacy-transformers<1.1.0,>=1.0.1' distribution was not
found and is required by the application", "The 'stanza<1.5.0,>=1.3.0'
distribution was not found and is required by the application", "The
'flair<1.0.0,>=0.6.0' distribution was not found and is required by the
application", "The 'ufal.udpipe' distribution was not found and is required by
the application", "The 'spacy-experimental==0.2.0' distribution was not found
and is required by the application", "The 'fastapi<0.62.0,>=0.61.1' distribution
was not found and is required by the application", "The 'aiofiles' distribution
was not found and is required by the application", "The
'uvicorn<0.12.0,>=0.11.6' distribution was not found and is required by the
application", "The 'spacy-huggingface_hub' distribution was not found and is
required by the application", "The 'plotext' distribution was not found and is
required by the application", "The 'spacy_ray' distribution was not found and is
required by the application", "The 'spacy-streamlit>=1.0.0a0' distribution was
not found and is required by the application", "The 'streamlit' distribution was
not found and is required by the application", "The 'wandb<0.20.0,>=0.12.4'
distribution was not found and is required by the application", "The
'floret<0.11.0,>=0.10.1' distribution was not found and is required by the
application", "The 'natto-py' distribution was not found and is required by the
application", "The 'spacy-streamlit>=1.0.0a0' distribution was not found and is
required by the application", "The 'google-cloud-storage' distribution was not
found and is required by the application", "The 'streamlit' distribution was not
found and is required by the application", "The 'presidio-analyzer==2.2.1'
distribution was not found and is required by the application", "The
'presidio-anonymizer==2.2.1' distribution was not found and is required by the
application", "The 'skweak==0.2.13' distribution was not found and is required
by the application", "The 'hmmlearn==0.2.6' distribution was not found and is
required by the application", "The 'nlpaug' distribution was not found and is
required by the application", 'ml-datasets 0.2.0 is installed but
ml_datasets==0.2.0a0 is required']
```

Reply:
This looks really hard to read. Can it be split into separate paragraphs?

Reply:
`wasabi` uses `textwrap.fill()` internally when preparing the message for printing, which by default seems to convert all space characters into whitespaces. Don't think there's a way around this?

Reply:
Maybe with each split into a separate message with `msg.text` or similar? I really don't think it needs to be the `repr` of a list of strings?